package.json

```diff
@@ -95,5 +98,8 @@
     "exclude": [
       "**/*.spec.js"
     ]
+  },
+  "lint-staged": {
+    "src/**/*.ts": "npm run fix -- --"
```

Comment:
How would you feel about having this run `lint` rather than `fix`, @meisekimiu? I feel a little strange about having a precommit hook that modifies the code I am committing; I would rather see an error.

Also, this doesn't appear to be working as I'd expect - if I have two linter errors, one staged and one unstaged, this is currently fixing both. As I understand it, the goal of `lint-staged` is to limit the scope of the linter to only staged files. I think this is happening because `fix` (and `lint`) are just wrappers around `run-s`, and are not properly passing their arguments down into the specific tools they call. It's not completely obvious to me how to fix this using [run-s](https://github.com/mysticatea/npm-run-all/blob/master/docs/run-s.md); it might be easier to break this [lint-staged configuration](https://www.npmjs.com/package/lint-staged#configuration) into a list and call eslint & prettier directly.

Reply:
I didn't realize that pre-commit checks can actually just error out and cancel a commit like that (although in hindsight it's kind of obvious)! I pushed up a commit that runs the lint check without autoformatting and it properly uses the linters directly so it works only on staged code.